site/public/js/server.js

```diff
@@ -825,7 +825,14 @@ function updateHomeworkExtensions(data) {
         cache: false,
         contentType: false,
         success: function(data) {
-            var json = JSON.parse(data);
+            //Will be a syntax error exception thrown if the user can't be found (<anonymous>)
+            try { 
+                var json = JSON.parse(data);
+            } catch(err){
+                var message ='<div class="inner-message alert alert-error" style="position: fixed;top: 40px;left: 50%;width: 40%;margin-left: -20%;" id="theid"><a class="fa fa-times message-close" onClick="removeMessagePopup(\'theid\');"></a><i class="fa fa-times-circle"></i>Invalid Student ID</div>';
+                $('#messages').append(message);
+                return;
+            }
```

Comment:
An invalid JSON object being returned by the endpoint should be fixed so that one a valid one (with a proper error message) is returned. JSON failing to parse should really only because the system threw an exception or something unexpected which causes invalid JSON (or returns an html page).

Reply:
You are right. I pushed two commits to reflect these changes.